src/components/course/data/service.js

```diff
@@ -29,8 +30,24 @@ export default class CourseService {
     ])
       .then((responses) => responses.map(response => response.data));
 
+    // Check for the course_run_key URL param and remove all other course run data
+    // if the given course run key is for an available course run.
+    const courseDetails = camelCaseObject(data[0]);
+    const { courseRunKey } = camelCaseObject(qs.parse(window.location.search));
```

Comment:
Rather than pulling `courseRunKey` off `window.location` within a service function, I think it might make more sense to pass `courseRunKey` into as an argument: `async fetchAllCourseData({ courseRunKey })`

In order to support that, you could do something like the following in `Course.jsx`:
```
const location = useLocation();

const courseRunKey = useMemo(
  () => {
    const queryParams = camelCaseObject(qs.parse(location.search));
    return queryParams.courseRunKey;
  },
  [location],
);

const [courseData, fetchError] = useAllCourseData({
  courseKey,
  enterpriseConfig,
  courseRunKey,
});
```

... and in the `useAllCourseData` hook itself, pass that `courseRunKey` into the service call:
```
const data = await courseService.fetchAllCourseData({
  courseRunKey,
});
```